### PR TITLE
fix: add input validation to prevent unbounded storage writes

### PR DESCRIPTION
**Pull request by mubarakcoded**
## Summary by CodeRabbit

* **New Features**
  * Early input validation to reject oversized or invalid group, welcome, and message data (names, descriptions, content, tags, events, admin keys, relays) before saving.

* **Bug Fixes**
  * Clearer validation error reporting when inputs exceed enforced limits.

* **Tests**
  * Added and adjusted tests to cover validation failures and boundary cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: crates/mdk-sqlite-storage/src/validation.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    if s.len() > max_length {
+        return Err(format!(
+            "{} exceeds maximum length of {} bytes (got {} bytes)",
+            field_name,
+            max_length,
+            s.len()
+        ));
+    }
+    Ok(())
+}
```
[the remark this answers is not in the document]
Comment (erskingardner) [reply crates/mdk-sqlite-storage/src/validation.rs]: Agree, let's make a new error variant in the Error enum.
